node/cn/handler.go

```diff
@@ -465,14 +465,24 @@ func (pm *ProtocolManager) handle(p Peer) error {
 }
 
 func (pm *ProtocolManager) processMsg(msgCh <-chan p2p.Msg, p Peer, addr common.Address, errCh chan<- error) {
-	for msg := range msgCh {
-		if err := pm.handleMsg(p, addr, msg); err != nil {
-			p.GetP2PPeer().Log().Error("ProtocolManager failed to handle message", "msg", msg, "err", err)
-			errCh <- err
-			return
+	_, fakeF := pm.fetcher.(*fetcher.FakeFetcher)
+	_, fakeD := pm.downloader.(*downloader.FakeDownloader)
+	if fakeD || fakeF {
+		p.GetP2PPeer().Log().Warn("ProtocolManager does not handle p2p messages", "fakeFetcher", fakeF, "fakeDownloader", fakeD)
+		for msg := range msgCh {
+			msg.Discard()
```

Comment:
There is no `return`. Is it ok?

Reply:
As we don't care what the message is, there's no need to handle the error from `Discard`.

Reply:
This function is executed as a goroutine.
If it is the fakeD/F case, there is no condition to return this function.
Am I misunderstanding?

Reply:
Oh, that can be a problem if the node is disconnected with the peer.
To recognize the error case, it seem that we need to check the error of `Discard`.

Reply:
The `processMsg` goroutine can be terminated when `msgCh` is closed, which is in `Handle` method with `defer close(channel)`.

Reply:
I missed that point. Thanks @jeongkyun-oh

Reply:
The goroutine will be closed when `msgCh` is closed.